components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/UserProfileAdmin.java

```diff
@@ -663,11 +671,11 @@ public void associateID(String idpID, String associatedID) throws UserProfileExc
      * @throws UserProfileException
      */
     public String getNameAssociatedWith(String idpID, String associatedID) throws UserProfileException {
-
+        
         try {
-            return UserProfileMgtDAO.getInstance().getUserAssociatedFor(CarbonContext.getThreadLocalCarbonContext()
-                    .getTenantId(), idpID, associatedID);
-        } catch (UserProfileException e) {
+            return getFederatedAssociationManager().getUserForTheFederatedAssociation(
```

Comment:
let's remove "The" part

Reply:
Fixed in 78f08459d2c398277e992c0c738443cfe4edad1b